openedx/frontend-app-course-authoring
# feat: add upload progress modal

## Description

The current implementation of video uploads notifies the user that there are uploads occurring but does not let the user cancel uploads not show the progress for individual uploads. This leads the user is suspense regarding which videos have completed the uploads. This PR add a modal the modal appears when the upload begins and disappears when upload is complete and progress bars as a visual representation to give the user a sense of how far along the upload is. This matches the upload flow of the legacy page with a new UI. This change impacts the Course Author.

## Supporting information

JIRA Ticket: [TNL-11629](https://2u-internal.atlassian.net/browse/TNL-11629)

>The new Video Uploads modal will make it clear to users that closing the Videos page will cancel the upload and that refreshing the page will cancel the upload. Additionally, the modal includes a progress bar so that user can see approximately how far along the upload is. 

## Testing instructions

1. Open the videos page
2. Click "Add video"
3. Select multiple videos to upload
4. Upload in-progress modal should appear
5. Confirm that clicking outside of the modal is disabled
6. Each of the videos that were selected should be listed
7. Confirm that the progress bars switch to failed
8. Modal closes automatically once all videos are processed
9. Should show upload error message for all failed uploads
10. Select another set of videos to upload
11. Upload in-progress should appear
12. Before videos finish, click "Cancel uploads"
13. Should show upload error message for failed uploads and cancelled uploads
